src/frontegg-auth/src/auth.rs

```diff
@@ -681,18 +768,40 @@ impl ValidatedClaims {
     }
 }
 
-fn validate_email(email: &str, expected_email: &str) -> Result<(), Error> {
-    // To match Frontegg, email addresses are compared case insensitively.
-    //
-    // NOTE(benesch): we could save some allocations by using `unicase::eq`
-    // here, but the `unicase` crate has had some critical correctness bugs that
-    // make it scary to use in such security-sensitive code.
+/// Reports whether a username is an email address.
+fn is_email(user: &str) -> bool {
+    // We don't need a sophisticated test here. We need a test that will return
+    // `true` for anything that can possibly be an email address, while also
+    // returning `false` for a large class of strings that can be used as names
+    // for service users.
     //
-    // See: https://github.com/seanmonstar/unicase/pull/39
-    if email.to_lowercase() != expected_email.to_lowercase() {
-        return Err(Error::WrongEmail);
+    // Checking for `@` balances the concerns. Every email address MUST have an
+    // `@` character. Disallowing `@` characters in service user names is an
+    // acceptable restriction.
+    user.contains('@')
+}
+
+fn validate_user(user: &str, expected_user: &str) -> Result<(), Error> {
+    // Impose a maximum length on user names for sanity.
+    if user.len() > MAX_USER_NAME_LENGTH {
```

Comment:
This could affect existing users. The chance seems low though. This seems like a thing to at least ask field eng about?

Reply:
Frontegg already enforces a maximum length on email addresses of 200 characters. So this limit is only really relevant for the new `user` metadata, and since that's new there aren't any existing names to worry about!